### PR TITLE
Code Changes

### PR DESCRIPTION
**Pull request by ParasuramanA**
1. Code changes are done in the file I worked on.
2. Added a custom hook separately for the handle changes
3. Made a separate component for data limit and search
4. Added them by keeping the promise all method as you said to list the add task page students
5. Added a tooltip and handle cancel

Discussion:
I have written a functionality to display the deadline date in the response data and if the deadline is today in the antd tag, it will be shown as the deadline. But it's not working properly, tell me if you get an error in it

### CODE REVIEW

File: package-lock.json
Comment (avinashdckap) [inline package-lock.json:1]: Remove this changes !

File: src/components/BreadCrumbs.jsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
-// Breadcrumbs.jsx
 import React from 'react';
-import { Breadcrumb } from 'antd';
 import { Link, useLocation } from 'react-router-dom';
+//external package
```
Comment (avinashdckap) [inline src/components/BreadCrumbs.jsx]: Just add a space between third party imports and custom imports and remove the //external package comments

File: src/components/CommonFilters.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          onChange={handleLimit}
+        >
+          <Option value="5">5</Option>
+          <Option value="10">10</Option>
+          <Option value="20">20</Option>
+          <Option value="30">30</Option>
+        </Select>
+        entries
+      </div>
+    </>
```
Comment (avinashdckap) [inline src/components/CommonFilters.jsx:37]: Include Total Records in this component

File: src/components/CommonFilters.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+//external packages here
+import Search from "antd/es/input/Search";
+import { Select } from "antd";
+import { Option } from "antd/es/mentions";
+
+const CommonFilter = ({ handleSearch, handleLimit, handleName ,handleChange}) => {
```
Comment (avinashdckap) [inline src/components/CommonFilters.jsx]: S missing in component name

File: src/components/CommonFilters.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        name={handleName}
+        onChange={handleChange}
+        onSearch={handleSearch}
+      />
+      <div className="fileter-category">
+        Show
+        <Select
+          size="small"
+          defaultValue="5"
+          style={{ width: 70, paddingLeft: 10, paddingRight: 5 }}
```
Comment (avinashdckap) [inline src/components/CommonFilters.jsx]: Inline styling remove it and add classnames

File: src/components/TableView.jsx [hunk 5]
```diff
@@ -1,34 +1,63 @@
...
 import React from 'react';
-import noDataFound from "../../public/images/no_data_found.svg";
-import { Button } from 'antd';
+//external packages here
+import moment from 'moment';
+import { Tag, Tooltip } from 'antd';
+//images here
+import noDataFound from "/images/no_data_found.svg";
 
+const TableComponent = ({ data, coulmnNameList, handleDelete, handleEdit }) => {
```
Comment (avinashdckap) [inline src/components/TableView.jsx]: Change it as Table View name

File: src/hooks/useAPI.jsx
Comment (avinashdckap) [inline src/hooks/useAPI.jsx:1]: Change useAPI  implementation to normal axios request

File: src/hooks/useForm.jsx
Comment (avinashdckap) [inline src/hooks/useForm.jsx:1]: Try react-hook-form for all the inputs in all pages

**Review by avinashdckap — CHANGES_REQUESTED** [review]

File: src/components/CommonFilters.jsx (hunk 2 above)
Comment (ParasuramanA) [reply src/components/CommonFilters.jsx:37]: added

File: src/components/CommonFilters.jsx (hunk 3 above)
Comment (ParasuramanA) [reply src/components/CommonFilters.jsx]: solved

File: src/components/TableView.jsx (hunk 5 above)
Comment (ParasuramanA) [reply src/components/TableView.jsx]: This component is renamed

**Review by avinashdckap — APPROVED** [review]

**Pull request merged**